0-common/igluctl/src/main/scala/com.snowplowanalytics.iglu/ctl/commands/Pull.scala

```diff
@@ -92,18 +96,24 @@ object Pull {
       else EitherT.leftT[IO, HttpResponse[String]](s"Unexpected status code ${response.code}. Response body: ${response.body}.")
     }
 
-  /**
-    * Parse the response from Iglu Server and write it under the specified output path.
-    * Perorms IO
+  /** Parse the response from Iglu Server
     *
     * @param response An [[HttpResponse]] returned by Iglu Server
+    * @return A list of [[SelfDescribingSchema]]
+    */
+  def parseResponse(response: HttpResponse[String]): List[Either[IgluParseError, SelfDescribingSchema[JValue]]] =
+    jacksonParse(response.body).extract[List[JValue]].map(igluParse[JValue](_))
+
+  /** Write [[SelfDescribingSchema]]s as files
+    * Performs IO
+    *
+    * @param schemas A list of [[SelfDescribingSchema]]
     * @param outputDir The path to write to
-    * @return A list of IO commands or Errors (if a schema fails validation), wrapped in an [[EitherT]]
+    * @return A list of IO commands or Errors (if a schema fails validation)
     */
-  def writeResponseAsSchemas(response: HttpResponse[String], outputDir: Path): EitherT[IO, Common.Error, List[String]] = {
-    val parsed = jacksonParse(response.body).extract[List[JValue]].map(igluParse[JValue](_).getOrElse(throw new RuntimeException("Invalid self-describing JSON schema"))).map {
-      schema => textFile(Paths.get(s"$outputDir/${schema.self.schemaKey.toPath}"), schema.asString)
-    }
-    parsed.map(_.write(true)).traverse(effect => EitherT(effect))
+  def writeSchemas(schemas: List[Either[IgluParseError, SelfDescribingSchema[JValue]]], outputDir: Path): EitherT[IO, Common.Error, List[String]] = {
```

Comment:
This is generally very good practice to have as simple as possile type of arguments. E.g. compare these:

```scala
def writeSchema(schema: SelfDescribingSchema[JValue], outputDir: Path)
// ^^ fairly obvious we're just writing a schema to disk - very atomic action
```

```scala
def writeSchemas(schemas: List[Either[IgluParseError, SelfDescribingSchema[JValue]]], outputDir: Path)
// ^^ we're traversing list... do something with errors (?)... and writing result to disk
```